examples/tools/scene_viewer.rs

```diff
@@ -0,0 +1,407 @@
+use bevy::{
+    asset::{AssetServerSettings, LoadState},
+    input::mouse::MouseMotion,
+    prelude::*,
+    render::{
+        camera::{Camera2d, Camera3d, CameraProjection},
+        primitives::{Aabb, Frustum, Sphere},
+    },
+    scene::InstanceId,
+};
+
+fn main() {
+    println!(
+        "
+Controls:
+    WSAD   - forward/back/strafe left/right
+    LShift - 'run'
+    E      - up
+    Q      - down
+    L      - animate light direction
+    U      - toggle shadows
+    5/6    - decrease/increase shadow projection width
+    7/8    - decrease/increase shadow projection height
+    9/0    - decrease/increase shadow projection near/far
+"
+    );
+    App::new()
+        .insert_resource(AmbientLight {
+            color: Color::WHITE,
+            brightness: 1.0 / 5.0f32,
+        })
+        .insert_resource(AssetServerSettings {
+            asset_folder: std::env::var("CARGO_MANIFEST_DIR").unwrap(),
+            watch_for_changes: true,
+        })
+        .insert_resource(WindowDescriptor {
+            title: "bevy scene viewer".to_string(),
+            ..default()
+        })
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup)
+        .add_system_to_stage(CoreStage::PreUpdate, scene_load_check)
+        .add_system_to_stage(CoreStage::PreUpdate, camera_spawn_check)
+        .add_system(camera_controller_check.label("camera_controller_check"))
```

Comment:
Please no string system labels. We can make a type for now, and then yeet it once Cart's PR merges.

Reply:
Done.